### ISSUE TITLE
can't use body tags in cljs tests (phantom js)

### ISSUE THREAD

**Issue by thedavidmeister** [hoplon/hoplon#175]
I see errors that look like this in tests where I try to call `hoplon.core/body`:

```
ERROR in (??current-page) (hoplon$core$remove_all_kids_BANG_@file:262:6)
Uncaught exception, not in assertion.
expected: nil
  actual: #object[TypeError TypeError: null is not an object (evaluating 'this$.hoplonKids = null')]
```

Looks like the issue is in `remove-all-kids!` in phantom js where `js/document` is `nil` and so there is no existing body tag to remove the kids from.

**Comment by flyboarder** [comment]
We really should be removing anything phantomjs related as the project is no longer maintained. In favour of using chrome-headless, this includes updating the prerender task to chrome/pupetteer .

**Comment by thedavidmeister** [comment]
@flyboarder i agree that we should shift to chrome headless.

FYI that this bug was cleaned up in 7.1 anyway as `hoplon.core/body` will defensively create a body tag if it doesn't already exist now.

**Comment by flyboarder** [comment]
For `7.2` should we move the defensive check to the `mksingleton` costructor? I can envision many js environments that might not have any of the singletons.

**Comment by thedavidmeister** [comment]
@flyboarder well document has to exist i think, but head might be missing i suppose?

the thing about making it generic is that these other low level tags might also need special handling:

```
(def body
 "Updates and returns the document's `body` element in place. Creates `body`
 if not exists."
 (mksingleton
  (do
   ; the body is not always set on the document (e.g. phantomjs tests)
   (when-not (.-body js/document)
    ; https://developer.mozilla.org/en-US/docs/Web/API/Document/body
    (set!
     (.-body js/document)
     (.createElement js/document "body")))
   (.-body js/document))))
```

you can see that we set the created body tag directly as a property of the document rather than using normal DOM manipulation.

it's hard to know how to make it generic without a concrete example from another environment

**Comment by thedavidmeister** [comment]
@flyboarder personally i think it's fine to provide special handling for `(.-body js/document)` as the browser already treats it as a special case.

i'm happy to close this off as-is if you are.

**Comment by flyboarder** [comment]
@thedavidmeister I think moving the defensive logic to the the body runtime would be better instead of shoving it into the document when hoplon is loaded.

**Comment by thedavidmeister** [comment]
@flyboarder i agree, that's what i did in https://github.com/hoplon/hoplon/pull/222

**Comment by flyboarder** [comment]
Fixed in latest `7.2-SNAPSHOT`.
